app/signals/apps/services/domain/signals/delete.py

```
@@ -0,0 +1,118 @@
+                DeleteSignalsService.delete_signal.delay(signal.id, batch_uuid, dry_run)
+            else:
+                DeleteSignalsService.delete_signal(signal.id, batch_uuid, dry_run)
+
+        return batch_uuid
+
+    @staticmethod
+    @app.task(prio=0)
+    def delete_signal(signal_id: int, batch_uuid: uuid.UUID = None, dry_run: bool = False):  # noqa C901
+        """
+        Delete a signal and associated data.
+
+        Args:
+            signal_id (int): The id of the signal to be deleted.
+            batch_uuid (UUID, optional): A unique identifier for the deletion batch.
+            dry_run (bool, optional): If True, no actual deletion occurs.
+        """
+        if dry_run:
+            return
+
+        try:
+            signal = Signal.objects.get(id=signal_id)
+        except Signal.DoesNotExist:
+            log.warning(f'Deleting Signal with id #{signal_id} went wrong, Signal no longer exists')
+            return
+
+        if signal.is_child:
+            note = (f'Parent signal was in state "{signal.parent.status.get_state_display()}" '
+                    f'for {(timezone.now() - signal.parent.status.created_at).days} days')
+        else:
+            note = (f'Signal was in state "{signal.status.get_state_display()}" '
+                    f'for {(timezone.now() - signal.status.created_at).days} days')
+
+        if signal.is_parent:
+            # Delete children before deleting the parent
+            for child_signal in signal.children.all():
+                DeleteSignalsService.delete_signal(child_signal.id, batch_uuid, dry_run)
+
+        try:
+            with transaction.atomic():
+                DeletedSignal.objects.create_from_signal(signal=signal, action='automatic', note=note,
+                                                         batch_uuid=batch_uuid)
+
+                try:
+                    delete_from_elastic(signal=signal)
+                except Exception:
+                    pass
```

Why ignore exceptions caused by deleting the signal from elasticsearch?